### PR TITLE
Exercise Space Interaction 2

### PR DESCRIPTION
**Pull request by ndittren**
* Interaction improvements 
* Interaction with holding and displaying correct hints. (Will probably disable the hint button for now)
* TO DO: Need to clear the input after completion of each step. 
* TO DO: Need to update immediately to complete when last step is reached. Right now it requires a refresh.

### CODE REVIEW

File: media/js/src/exerciseSpace.tsx [hunk 1]
```diff
@@ -25,39 +26,74 @@ export const ExerciseSpace: React.FC = () => {
...
+    };
+    async function fetchHints() {
+        const hintData: HintData = {
+            next_expr: '',
+            rule: '',
+            step_list: [''],
+            answer: ''
+        };
+        //Only if we don't already have hints
+        if(hintButtonCount === 0) {
```
Comment (sdreher) [inline media/js/src/exerciseSpace.tsx:64]: Check the eslint settings here - should be a space between the conditional and the (, e.g. `if (hintButtonCount === 0`

File: media/js/src/exerciseSpace.tsx [hunk 2]
```diff
@@ -25,39 +26,74 @@ export const ExerciseSpace: React.FC = () => {
...
+                hintData['next_expr'] = nextStep;
+                hintData['rule'] = nextRule;
+                hintData['step_list'] = [lastCorrectStep];
+                hintData['answer'] = statement.answer;
+                // eslint-disable-next-line max-len
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                const toolsData: Tools = await getHints(hintData);
+                setHint([toolsData.hintRule, toolsData.hintExpression]);
+            }
+        } else {
```
Comment (sdreher) [inline media/js/src/exerciseSpace.tsx:93]: No need for this `else` statement.

File: media/js/src/exerciseSpace.tsx [hunk 3]
```diff
@@ -122,42 +147,21 @@ export const ExerciseSpace: React.FC = () => {
     };
     const isIncomplete = status[questionStatus] !== 'complete';
     const showSolutionBtn = stepList.length >= 2;
+    // eslint-disable-next-line max-len
+    const quesText: string = (statement.answer == 'F') || (statement.answer == 'T')
```
Comment (sdreher) [inline media/js/src/exerciseSpace.tsx:151]: `===` here

File: media/js/src/solutionStep.tsx [hunk 4]
```diff
@@ -59,38 +63,107 @@ export const SolutionStep: React.FC<SolutionStepProps> = (
...
         evt.preventDefault();
-        if(!statementInput) {
+        if(hintButtonCount === 2) {
+            setHintButtonCount(0);
+        }
+        if(!nextStep) {
             setError('Please enter a statement.');
+        } else {
+            setError('');
+            void validateStep();
```
Comment (sdreher) [inline media/js/src/solutionStep.tsx:141]: Is the `void` needed here?

Comment (ndittren) [reply media/js/src/solutionStep.tsx:141]: I believe void is best here since that async function isn't returning anything.

**Pull request merged**